pkgs/os-specific/linux/upower/default.nix

```diff
@@ -17,6 +17,7 @@
 , libusb1
 , glib
 , gettext
+, polkit
 , systemd
 , nixosTests
```

Comment:
What are the test failures?

Reply:
Here's the report from the build log:
<details>
<summary>nix log /nix/store/1wlq4hngar4y4dgglgma1z1w5w5gl3f7-upower-1.90.5.drv</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/28x69whchcy9ch54s8byxva1wlxlqs5k-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/n394mrdf7hhlw2pf5211lfg8wb27r7p7-installed-tests-path.patch
patching file meson_options.txt
patching file src/meson.build
Hunk #1 succeeded at 88 (offset 3 lines).
Hunk #2 succeeded at 151 (offset 3 lines).
patching file src/upower-integration.test.in
patching script interpreter paths in src/linux/integration-test.py
src/linux/integration-test.py: interpreter directive changed from "#!/usr/bin/python3 -u" to "/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3 -u"
patching script interpreter paths in src/linux/unittest_inspector.py
src/linux/unittest_inspector.py: interpreter directive changed from "#! /usr/bin/env python3" to "/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3"
substituteStream() in derivation upower-1.90.5: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'src/linux/integration-test.py')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
mesonConfigurePhase flags: --prefix=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5 --libdir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/lib --libexecdir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/libexec --bindir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/bin --sbindir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/sbin --includedir=/nix/store/1i818q32j0yqzcpx29av26wknb6zigq6-upower-1.90.5-dev/include --mandir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share/man --infodir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share/info --localedir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain --localstatedir=/var --sysconfdir=/etc -Dos_backend=linux -Dsystemdsystemunitdir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/etc/systemd/system -Dudevrulesdir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/lib/udev/rules.d -Dudevhwdbdir=/nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/lib/udev/hwdb.d -Dintrospection=enabled -Dgtk-doc=true -Dinstalled_test_prefix=/nix/store/ib6pz9ywn3b17v56nam9bm2p09m8r1gx-upower-1.90.5-installedTests
[1mThe Meson build system[0m
Version: 1.5.1
Source dir: [1m/build/source[0m
Build dir: [1m/build/source/build[0m
Build type: [1mnative build[0m
Project name: [1mupower[0m
Project version: [1m1.90.5[0m
C compiler for the host machine: [1mgcc[0m (gcc 13.3.0 "gcc (GCC) 13.3.0")
C linker for the host machine: [1mgcc[0m ld.bfd 2.42
Host machine cpu family: [1mx86_64[0m
Host machine cpu: [1mx86_64[0m
Compiler for C supports arguments -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_66: [1;32mYES[0m 
Compiler for C supports arguments -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_66: [1;32mYES[0m 
Found pkg-config: [1;32mYES[0m [1m(/nix/store/02vzklfgfpaf457rq927fq3jj5hn133f-pkg-config-wrapper-0.29.2/bin/pkg-config)[0m [1;34m0.29.2[0m
Run-time dependency [1mglib-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m
Run-time dependency [1mgobject-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m
Run-time dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m
Run-time dependency [1mgio-unix-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m
Library [1mm[0m found: [1;32mYES[0m
Run-time dependency [1mpolkit-gobject-1[0m found: [1;32mYES[0m [36m124[0m
Program [1mxsltproc[0m found: [1;32mYES[0m (/nix/store/g9aj2skqqhz6frsi5hdhc5pmrwwiip4d-libxslt-1.1.42-bin/bin/xsltproc)
Run-time dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.80.1[0m
Run-time dependency [1mgudev-1.0[0m found: [1;32mYES[0m [36m238[0m
Run-time dependency [1mlibimobiledevice-1.0[0m found: [1;32mYES[0m [36m1.3.0-unstable-2024-05-20[0m
Run-time dependency [1mlibplist-2.0[0m found: [1;32mYES[0m [36m2.6.0[0m
Configuring [1mconfig.h[0m using configuration
Program [1mmsgfmt[0m found: [1;32mYES[0m (/nix/store/p1mg5zzs2bk762pamss9cdshdsr0n19p-gettext-0.21.1/bin/msgfmt)
Program [1mmsginit[0m found: [1;32mYES[0m (/nix/store/p1mg5zzs2bk762pamss9cdshdsr0n19p-gettext-0.21.1/bin/msginit)
Program [1mmsgmerge[0m found: [1;32mYES[0m (/nix/store/p1mg5zzs2bk762pamss9cdshdsr0n19p-gettext-0.21.1/bin/msgmerge)
Program [1mxgettext[0m found: [1;32mYES[0m (/nix/store/p1mg5zzs2bk762pamss9cdshdsr0n19p-gettext-0.21.1/bin/xgettext)
Dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m [1;34m(cached)[0m
Program [1m/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/bin/gdbus-codegen[0m found: [1;32mYES[0m (/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/bin/gdbus-codegen)
Dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m [1;34m(cached)[0m
Program [1m/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/bin/gdbus-codegen[0m found: [1;32mYES[0m (/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/bin/gdbus-codegen)
Dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.80.4[0m [1;34m(cached)[0m
Program [1m/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/bin/gdbus-codegen[0m found: [1;32mYES[0m (/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/bin/gdbus-codegen)
Configuring [1mup-version.h[0m using configuration
Dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.80.1[0m [1;34m(cached)[0m
Dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.80.1[0m [1;34m(cached)[0m
Program [1m/nix/store/2ivbi3zdfjxyasjbd4hz787kjp5ypsmj-gobject-introspection-1.80.1-dev/bin/g-ir-scanner[0m found: [1;32mYES[0m (/nix/store/2ivbi3zdfjxyasjbd4hz787kjp5ypsmj-gobject-introspection-1.80.1-dev/bin/g-ir-scanner)
Dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.80.1[0m [1;34m(cached)[0m
Program [1m/nix/store/2ivbi3zdfjxyasjbd4hz787kjp5ypsmj-gobject-introspection-1.80.1-dev/bin/g-ir-compiler[0m found: [1;32mYES[0m (/nix/store/2ivbi3zdfjxyasjbd4hz787kjp5ypsmj-gobject-introspection-1.80.1-dev/bin/g-ir-compiler)
Configuring [1morg.freedesktop.UPower.service[0m using configuration
Configuring [1morg.freedesktop.UPower.conf[0m using configuration
Configuring [1mupower.service[0m using configuration
Program [1mpython3[0m found: [1;32mYES[0m (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3)
Program [1mlinux/unittest_inspector.py[0m found: [1;32mYES[0m (/build/source/src/linux/unittest_inspector.py)
Configuring [1mupower-integration.test[0m using configuration
Configuring [1mversion.xml[0m using configuration
Program [1mgtkdoc-scan[0m found: [1;32mYES[0m (/nix/store/gbi9fwqdsps73zrcjwb8z8hvyk0nb465-gtk-doc-1.34.0/bin/gtkdoc-scan)
Program [1mgtkdoc-scangobj[0m found: [1;32mYES[0m (/nix/store/gbi9fwqdsps73zrcjwb8z8hvyk0nb465-gtk-doc-1.34.0/bin/gtkdoc-scangobj)
Program [1mgtkdoc-mkdb[0m found: [1;32mYES[0m (/nix/store/gbi9fwqdsps73zrcjwb8z8hvyk0nb465-gtk-doc-1.34.0/bin/gtkdoc-mkdb)
Program [1mgtkdoc-mkhtml[0m found: [1;32mYES[0m (/nix/store/gbi9fwqdsps73zrcjwb8z8hvyk0nb465-gtk-doc-1.34.0/bin/gtkdoc-mkhtml)
Program [1mgtkdoc-fixxref[0m found: [1;32mYES[0m (/nix/store/gbi9fwqdsps73zrcjwb8z8hvyk0nb465-gtk-doc-1.34.0/bin/gtkdoc-fixxref)
[1mMessage:[0m 
UPower 1.90.5
System Paths
  prefix:         /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5
  libdir:         lib
  libexecdir:     /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/libexec
  bindir:         /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/bin
  sbindir:        /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/sbin
  datadir:        /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share
  sysconfdir:     /etc
  localstatedir:  /var
  historydir:     /var/lib/upower

Features
  Backend:                  linux
  libimobiledevice support: true
  Building api docs:        true
  Building man pages:       true

Build targets in project: [1m35[0m
[1mNOTICE:[0m Future-deprecated features used:
 * 1.1.0: {'"boolean option" keyword argument "value" of type str'}

upower [36m1.90.5[0m

  [1mUser defined options[0m
    auto_features        : enabled
    bindir               : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/bin
    buildtype            : plain
    includedir           : /nix/store/1i818q32j0yqzcpx29av26wknb6zigq6-upower-1.90.5-dev/include
    infodir              : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share/info
    libdir               : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/lib
    libexecdir           : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/libexec
    localedir            : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share/locale
    localstatedir        : /var
    mandir               : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/share/man
    prefix               : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5
    sbindir              : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/sbin
    sysconfdir           : /etc
    wrap_mode            : nodownload
    gtk-doc              : true
    installed_test_prefix: /nix/store/ib6pz9ywn3b17v56nam9bm2p09m8r1gx-upower-1.90.5-installedTests
    introspection        : enabled
    os_backend           : linux
    systemdsystemunitdir : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/etc/systemd/system
    udevhwdbdir          : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/lib/udev/hwdb.d
    udevrulesdir         : /nix/store/fl75cckwacz36c97ljh3nrnk75wry54h-upower-1.90.5/lib/udev/rules.d

Found ninja-1.12.1 at /nix/store/m0g218hn4rbq2wj7szzg8a112alq1qvc-ninja-1.12.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j12
[1/61] Building translation po/pl/LC_MESSAGES/upower-pl.mo
[2/61] Building translation po/fr/LC_MESSAGES/upower-fr.mo
[3/61] Building translation po/it/LC_MESSAGES/upower-it.mo
[4/61] Building translation po/sv/LC_MESSAGES/upower-sv.mo
[5/61] Generating doc/org.freedesktop.UPower.ref.xml with a custom command
[6/61] Generating doc/org.freedesktop.UPower.Device.ref.xml with a custom command
[7/61] Generating doc/org.freedesktop.UPower.KbdBacklight.ref.xml with a custom command
[8/61] Generating gdbus source dbus/up-device-generated.c
[9/61] Merging translations for policy/org.freedesktop.upower.policy
[10/61] Generating gdbus source dbus/up-kbd-backlight-generated.c
[11/61] Generating gdbus header dbus/up-kbd-backlight-generated.h
[12/61] Generating gdbus source dbus/up-daemon-generated.c
[13/61] Generating gdbus header dbus/up-device-generated.h
[14/61] Generating gdbus header dbus/up-daemon-generated.h
[15/61] Generating doc/man/upowerd.8 with a custom command
Note: Writing upowerd.8
[16/61] Generating doc/man/upower.1 with a custom command
Note: Writing upower.1
[17/61] Generating doc/man/UPower.7 with a custom command
Note: Writing UPower.7
[18/61] Compiling C object src/dummy/libupshared.a.p/up-native.c.o
[19/61] Compiling C object tools/upower.p/up-tool.c.o
[20/61] Compiling C object libupower-glib/libupower-glib.so.3.1.0.p/up-types.c.o
[21/61] Compiling C object libupower-glib/libupower-glib.so.3.1.0.p/up-stats-item.c.o
[22/61] Compiling C object libupower-glib/libupower-glib.so.3.1.0.p/up-history-item.c.o
[23/61] Compiling C object src/dummy/libupshared.a.p/up-backend.c.o
[24/61] Compiling C object libupower-glib/libupower-glib.so.3.1.0.p/up-client.c.o
[25/61] Compiling C object src/linux/libupshared.a.p/up-device-supply.c.o
[26/61] Compiling C object dbus/liblibupower-dbus.a.p/meson-generated_.._up-kbd-backlight-generated.c.o
[27/61] Compiling C object src/linux/libupshared.a.p/up-device-bluez.c.o
[28/61] Compiling C object libupower-glib/libupower-glib.so.3.1.0.p/up-device.c.o
[29/61] Compiling C object src/linux/libupshared.a.p/up-device-hid.c.o
[30/61] Compiling C object src/linux/libupshared.a.p/up-device-supply-battery.c.o
[31/61] Compiling C object src/linux/libupshared.a.p/up-device-wup.c.o
[32/61] Compiling C object src/linux/libupshared.a.p/up-input.c.o
[33/61] Compiling C object src/linux/libupshared.a.p/up-native.c.o
[34/61] Compiling C object dbus/liblibupower-dbus.a.p/meson-generated_.._up-daemon-generated.c.o
[35/61] Compiling C object src/libupowerd-private.a.p/up-config.c.o
[36/61] Compiling C object src/linux/libupshared.a.p/up-device-idevice.c.o
[37/61] Compiling C object src/libupowerd-private.a.p/up-enumerator.c.o
[38/61] Compiling C object src/linux/libupshared.a.p/up-enumerator-udev.c.o
[39/61] Linking static target src/dummy/libupshared.a
[40/61] Compiling C object src/linux/libupshared.a.p/up-backend.c.o
[41/61] Linking static target src/linux/libupshared.a
[42/61] Compiling C object src/libupowerd-private.a.p/up-device-list.c.o
[43/61] Compiling C object src/libupowerd-private.a.p/up-common.c.o
[44/61] Compiling C object src/libupowerd-private.a.p/up-device-battery.c.o
[45/61] Compiling C object src/libupowerd-private.a.p/up-kbd-backlight.c.o
[46/61] Compiling C object src/libupowerd-private.a.p/up-device.c.o
[47/61] Compiling C object src/upowerd.p/up-main.c.o
[48/61] Compiling C object src/libupowerd-private.a.p/up-daemon.c.o
[49/61] Compiling C object src/libupowerd-private.a.p/up-polkit.c.o
[50/61] Compiling C object src/libupowerd-private.a.p/up-history.c.o
[51/61] Linking static target src/libupowerd-private.a
[52/61] Compiling C object src/up_self_test.p/up-self-test.c.o
[53/61] Compiling C object dbus/liblibupower-dbus.a.p/meson-generated_.._up-device-generated.c.o
[54/61] Linking static target dbus/liblibupower-dbus.a
[55/61] Linking target libupower-glib/libupower-glib.so.3.1.0
[56/61] Generating symbol file libupower-glib/libupower-glib.so.3.1.0.p/libupower-glib.so.3.1.0.symbols
[57/61] Linking target tools/upower
[58/61] Linking target src/up_self_test
[59/61] Linking target src/upowerd
[60/61] Generating libupower-glib/UPowerGlib-1.0.gir with a custom command (wrapped by meson to set env)
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gutils.h:378: syntax error, unexpected INTEGER, expecting identifier in '  if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 0); if (nth_bit < -1) G_PASTE (_g_boolean_var_, 0) = 1; else G_PASTE (_g_boolean_var_, 0) = 0; G_PASTE (_g_boolean_var_, 0); }), 0)))' at '0'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gutils.h:393: syntax error, unexpected INTEGER, expecting identifier in '  if (nth_bit < 0 || (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 1); if (nth_bit > 8 * 8) G_PASTE (_g_boolean_var_, 1) = 1; else G_PASTE (_g_boolean_var_, 1) = 0; G_PASTE (_g_boolean_var_, 1); }), 0)))' at '1'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gutils.h:408: syntax error, unexpected INTEGER, expecting identifier in '  return (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 2); if (number) G_PASTE (_g_boolean_var_, 2) = 1; else G_PASTE (_g_boolean_var_, 2) = 0; G_PASTE (_g_boolean_var_, 2); }), 1)) ?' at '2'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gstring.h:198: syntax error, unexpected INTEGER, expecting identifier in '  if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 3); if (gstring != ' at '3'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gstring.h:220: syntax error, unexpected INTEGER, expecting identifier in '  if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 4); if (gstring == ' at '4'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gstring.h:223: syntax error, unexpected INTEGER, expecting identifier in '  if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 5); if (val == ' at '5'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gstring.h:231: syntax error, unexpected INTEGER, expecting identifier in '  if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 6); if (gstring->len + len_unsigned < gstring->allocated_len) G_PASTE (_g_boolean_var_, 6) = 1; else G_PASTE (_g_boolean_var_, 6) = 0; G_PASTE (_g_boolean_var_, 6); }), 1)))' at '6'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gstring.h:234: syntax error, unexpected IF in '      if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 7); if (val + len_unsigned <= end || val > end + len_unsigned) G_PASTE (_g_boolean_var_, 7) = 1; else G_PASTE (_g_boolean_var_, 7) = 0; G_PASTE (_g_boolean_var_, 7); }), 1)))' at 'if'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/glib/gstring.h:234: syntax error, unexpected INTEGER, expecting identifier in '      if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 7); if (val + len_unsigned <= end || val > end + len_unsigned) G_PASTE (_g_boolean_var_, 7) = 1; else G_PASTE (_g_boolean_var_, 7) = 0; G_PASTE (_g_boolean_var_, 7); }), 1)))' at '7'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h:804: syntax error, unexpected INTEGER, expecting identifier in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 8); if ((((g_type_check_instance_is_fundamentally_a ((GTypeInstance*) ((weak_pointer)), (((GType) ((20) << (2))))))))) G_PASTE (_g_boolean_var_, 8) = 1; else G_PASTE (_g_boolean_var_, 8) = 0; G_PASTE (_g_boolean_var_, 8); }), 1)) ; else g_assertion_message (((gchar*) 0), "/nix/store/6R6RZV2V3X8MH2KI391GVC9Y9954YZLG-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h", 804, ((const char*) (__func__)), "'" "G_IS_OBJECT (weak_pointer)" "' should be TRUE"); } while (0);' at '8'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h:804: syntax error, unexpected ')', expecting identifier or '(' in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 8); if ((((g_type_check_instance_is_fundamentally_a ((GTypeInstance*) ((weak_pointer)), (((GType) ((20) << (2))))))))) G_PASTE (_g_boolean_var_, 8) = 1; else G_PASTE (_g_boolean_var_, 8) = 0; G_PASTE (_g_boolean_var_, 8); }), 1)) ; else g_assertion_message (((gchar*) 0), "/nix/store/6R6RZV2V3X8MH2KI391GVC9Y9954YZLG-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h", 804, ((const char*) (__func__)), "'" "G_IS_OBJECT (weak_pointer)" "' should be TRUE"); } while (0);' at ')'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h:807: syntax error, unexpected INTEGER, expecting identifier in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 9); if ((weak_pointer) == ' at '9'
/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h:807: syntax error, unexpected ')', expecting identifier or '(' in '# 807 "/nix/store/6r6rzv2v3x8mh2ki391gvc9y9954yzlg-glib-2.80.4-dev/include/glib-2.0/gobject/gobject.h"' at ')'
libupower-glib/up-version.h:22: Warning: UPowerGlib: multiple comment blocks documenting 'SECTION:up-version:' identifier (already seen at up-version.h:22).
[61/61] Generating libupower-glib/UPowerGlib-1.0.typelib with a custom command
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
ninja: Entering directory `/build/source/build'
ninja: no work to do.
[1-12/75] 🌑 Tests.test_battery_overfull                                                                       0/80s
[K 1/75 Tests.test_battery_charge_limit_multiple_batteries                                       [1;31mFAIL          [0m  0.94s   exit status 1
[1;31m>>> [0mLD_LIBRARY_PATH=/build/source/build/libupower-glib:/build/source/build/libupower-glib MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 GI_TYPELIB_PATH=/build/source/build/libupower-glib:/nix/store/pjxrn2wn0sn533p48jz5qxgjld84hn3i-glib-2.80.4/lib/girepository-1.0:/nix/store/vscjxy29f1w1z36ah69x2v37fvf8p1nm-gobject-introspection-1.80.1/lib/girepository-1.0:/nix/store/m4s69jvb62d9adgj0jxbw5qq3j3b4b9b-libgudev-238/lib/girepository-1.0:/nix/store/5l380ssy468jw4czvjyp25sb739hiczr-polkit-124/lib/girepository-1.0:/nix/store/5ryn154cpg2vw1bn5ybmrcl4mcsjsphq-umockdev-0.18.3/lib/girepository-1.0 MESON_TEST_ITERATION=1 top_builddir=/build/source/build ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 MALLOC_PERTURB_=185 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3 /build/source/build/../src/linux/integration-test.py Tests.test_battery_charge_limit_multiple_batteries
 ✀  
[1;36mstdout:[0m
Testing binaries from local build tree
1725997136.985 GetAll /org/freedesktop/login1 org.freedesktop.login1.Manager
1725997136.986 Inhibit "sleep" "UPower" "Pause device polling" "delay"
[32mTI:19:38:56	[34mAcquired inhibitor lock (7, delay)
[0m[32mTI:19:38:56	[34mStarting upowerd version 1.90.5
[0m[32mTI:19:38:56	[34mcannot find a keyboard backlight
[0m[32mTI:19:38:56	[34mExported UpDevice with path /org/freedesktop/UPower/devices/DisplayDevice
[0m[32mTI:19:38:56	[34mdaemon now coldplug
[0m[32mTI:19:38:56	[34mregistering subsystem : input
[0m[32mTI:19:38:56	[34mregistering subsystem : power_supply
[0m[32mTI:19:38:56	[34mReceived uevent add on device /sys/devices/BAT0
[0m[32mTI:19:38:56	[34musing present voltage (alternate)
[0m[32mTI:19:38:56	[34mfailed to read battery charge threshold: Failed to open file “/build/upower-history-vkcxp3rs/charging-threshold-status”: No such file or directory
[0m[32mTI:19:38:56	[34musing id: 80
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-rate-80.dat as file does not exist
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-charge-80.dat as file does not exist
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-time-full-80.dat as file does not exist
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-time-empty-80.dat as file does not exist
[0m[32mTI:19:38:56	[34msaving in 600 seconds
[0m[32mTI:19:38:56	[34mdevice BAT0 has unknown state, not saving history
[0m[32mTI:19:38:56	[34munknown_poll: setting up fast re-poll
[0m[32mTI:19:38:56	[34madded native-path: BAT0
[0m[32mTI:19:38:56	[34mExported UpDevice with path /org/freedesktop/UPower/devices/battery_BAT0
[0m[32mTI:19:38:56	[34mdevice BAT0 has parent id: (null)
[0m[32mTI:19:38:56	[34mGot new device from udev enumerator: 0x45ee10
[0m[32mTI:19:38:56	[34madded BAT0
[0m[32mTI:19:38:56	[34memitting added: /org/freedesktop/UPower/devices/battery_BAT0
[0m[32mTI:19:38:56	[34mReceived uevent add on device /sys/devices/BAT1
[0m[32mTI:19:38:56	[34musing present voltage (alternate)
[0m[32mTI:19:38:56	[34mfailed to read battery charge threshold: Failed to open file “/build/upower-history-vkcxp3rs/charging-threshold-status”: No such file or directory
[0m[32mTI:19:38:56	[34musing id: 80
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-rate-80.dat as file does not exist
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-charge-80.dat as file does not exist
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-time-full-80.dat as file does not exist
[0m[32mTI:19:38:56	[34mfailed to get data from /build/upower-history-vkcxp3rs/history-time-empty-80.dat as file does not exist
[0m[32mTI:19:38:56	[34msaving in 600 seconds
[0m[32mTI:19:38:56	[34mdevice BAT1 has unknown state, not saving history
[0m[32mTI:19:38:56	[34munknown_poll: setting up fast re-poll
[0m[32mTI:19:38:56	[34madded native-path: BAT1
[0m[32mTI:19:38:56	[34mExported UpDevice with path /org/freedesktop/UPower/devices/battery_BAT1
[0m[32mTI:19:38:56	[34mdevice BAT1 has parent id: (null)
[0m[32mTI:19:38:56	[34mGot new device from udev enumerator: 0x461af0
[0m[32mTI:19:38:56	[34madded BAT1
[0m[32mTI:19:38:56	[34memitting added: /org/freedesktop/UPower/devices/battery_BAT1
[0m[32mTI:19:38:56	[34mregistering subsystem : usb
[0m[32mTI:19:38:56	[34mregistering subsystem : usbmisc
[0m[32mTI:19:38:56	[34mregistering subsystem : sound
[0m[32mTI:19:38:56	[34mCalculating percentage and time to full/to empty for 2 batteries
[0m[32mTI:19:38:56	[34mon_battery = yes
[0m[32mTI:19:38:56	[34mwarning_level = none
[0m[32mTI:19:38:56	[34mBlueZ disappeared
[0m[32mTI:19:38:56	[34mdaemon now not coldplug
[0m[32mTI:19:38:57	[34mError on Polkit check authority: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.PolicyKit1 was not provided by any .service files
[0m[32mTI:19:38:57	[34mHandling SIGTERM
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-rate-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-charge-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-time-full-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-time-empty-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-rate-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-charge-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-time-full-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-vkcxp3rs/history-time-empty-80.dat
[0m
[1;36mstderr:[0m
E
======================================================================
ERROR: test_battery_charge_limit_multiple_batteries (__main__.Tests.test_battery_charge_limit_multiple_batteries)
Battery with charge limits with multiple batteries
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/build/../src/linux/integration-test.py", line 1055, in test_battery_charge_limit_multiple_batteries
    self.enable_charge_limits(bat0_up, True)
  File "/build/source/build/../src/linux/integration-test.py", line 297, in enable_charge_limits
    self.dbus.call_sync(UP, device,
gi.repository.GLib.GError: g-io-error-quark: GDBus.Error:org.freedesktop.UPower.GeneralError: Operation is not allowed. (36)

----------------------------------------------------------------------
Ran 1 test in 0.447s

FAILED (errors=1)


[2-12/75] 🌒 Tests.test_battery_overfull                                                                       0/80s
[K[2-13/75] 🌓 Tests.test_battery_overfull                                                                       0/80s
[K 2/75 Tests.test_battery_broken_name                                                           [1;32mOK            [0m  0.97s
[3-13/75] 🌔 Tests.test_battery_overfull                                                                       0/80s
[K[3-14/75] 🌕 Tests.test_battery_overfull                                                                       0/80s
[K 3/75 Tests.test_absent_device_battery                                                         [1;32mOK            [0m  0.98s
[4-14/75] 🌖 Tests.test_battery_overfull                                                                       0/80s
[K[4-15/75] 🌗 Tests.test_battery_overfull                                                                       0/80s
[K[4-15/75] 🌘 Tests.test_battery_zero_power_draw                                                                0/80s
[K 4/75 Tests.test_battery_capacity_and_charge                                                   [1;32mOK            [0m  1.03s
[5-15/75] 🌑 Tests.test_battery_zero_power_draw                                                                0/80s
[K[5-16/75] 🌒 Tests.test_battery_zero_power_draw                                                                0/80s
[K 5/75 Tests.test_battery_charge                                                                [1;32mOK            [0m  1.04s
[6-16/75] 🌓 Tests.test_battery_zero_power_draw                                                                0/80s
[K[6-17/75] 🌔 Tests.test_battery_zero_power_draw                                                                0/80s
[K 6/75 Tests.test_battery_charge_threshold_unsupported                                          [1;31mFAIL          [0m  1.04s   exit status 1
[1;31m>>> [0mLD_LIBRARY_PATH=/build/source/build/libupower-glib:/build/source/build/libupower-glib MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 GI_TYPELIB_PATH=/build/source/build/libupower-glib:/nix/store/pjxrn2wn0sn533p48jz5qxgjld84hn3i-glib-2.80.4/lib/girepository-1.0:/nix/store/vscjxy29f1w1z36ah69x2v37fvf8p1nm-gobject-introspection-1.80.1/lib/girepository-1.0:/nix/store/m4s69jvb62d9adgj0jxbw5qq3j3b4b9b-libgudev-238/lib/girepository-1.0:/nix/store/5l380ssy468jw4czvjyp25sb739hiczr-polkit-124/lib/girepository-1.0:/nix/store/5ryn154cpg2vw1bn5ybmrcl4mcsjsphq-umockdev-0.18.3/lib/girepository-1.0 MESON_TEST_ITERATION=1 MALLOC_PERTURB_=198 top_builddir=/build/source/build ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3 /build/source/build/../src/linux/integration-test.py Tests.test_battery_charge_threshold_unsupported
 ✀  
[1;36mstdout:[0m
Testing binaries from local build tree
1725997137.042 GetAll /org/freedesktop/login1 org.freedesktop.login1.Manager
1725997137.042 Inhibit "sleep" "UPower" "Pause device polling" "delay"
[32mTI:19:38:57	[34mAcquired inhibitor lock (7, delay)
[0m[32mTI:19:38:57	[34mStarting upowerd version 1.90.5
[0m[32mTI:19:38:57	[34mcannot find a keyboard backlight
[0m[32mTI:19:38:57	[34mExported UpDevice with path /org/freedesktop/UPower/devices/DisplayDevice
[0m[32mTI:19:38:57	[34mdaemon now coldplug
[0m[32mTI:19:38:57	[34mregistering subsystem : input
[0m[32mTI:19:38:57	[34mregistering subsystem : power_supply
[0m[32mTI:19:38:57	[34mReceived uevent add on device /sys/devices/BAT0
[0m[32mTI:19:38:57	[34musing present voltage (alternate)
[0m[32mTI:19:38:57	[34mfailed to read battery charge threshold: Failed to open file “/build/upower-history-y6dxbp2n/charging-threshold-status”: No such file or directory
[0m[32mTI:19:38:57	[34musing id: test-80
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-y6dxbp2n/history-rate-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-y6dxbp2n/history-charge-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-y6dxbp2n/history-time-full-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-y6dxbp2n/history-time-empty-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34msaving in 600 seconds
[0m[32mTI:19:38:57	[34mdevice BAT0 has unknown state, not saving history
[0m[32mTI:19:38:57	[34munknown_poll: setting up fast re-poll
[0m[32mTI:19:38:57	[34madded native-path: BAT0
[0m[32mTI:19:38:57	[34mExported UpDevice with path /org/freedesktop/UPower/devices/battery_BAT0
[0m[32mTI:19:38:57	[34mdevice BAT0 has parent id: (null)
[0m[32mTI:19:38:57	[34mGot new device from udev enumerator: 0x45e840
[0m[32mTI:19:38:57	[34madded BAT0
[0m[32mTI:19:38:57	[34memitting added: /org/freedesktop/UPower/devices/battery_BAT0
[0m[32mTI:19:38:57	[34mregistering subsystem : usb
[0m[32mTI:19:38:57	[34mregistering subsystem : usbmisc
[0m[32mTI:19:38:57	[34mregistering subsystem : sound
[0m[32mTI:19:38:57	[34mon_battery = yes
[0m[32mTI:19:38:57	[34mwarning_level = none
[0m[32mTI:19:38:57	[34mdaemon now not coldplug
[0m[32mTI:19:38:57	[34mBlueZ disappeared
[0m[32mTI:19:38:57	[34mError on Polkit check authority: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.PolicyKit1 was not provided by any .service files
[0m[32mTI:19:38:57	[34mHandling SIGTERM
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-y6dxbp2n/history-rate-test-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-y6dxbp2n/history-charge-test-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-y6dxbp2n/history-time-full-test-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-y6dxbp2n/history-time-empty-test-80.dat
[0m
[1;36mstderr:[0m
F
======================================================================
FAIL: test_battery_charge_threshold_unsupported (__main__.Tests.test_battery_charge_threshold_unsupported)
Battery with only end_threshold supported
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/build/../src/linux/integration-test.py", line 1163, in test_battery_charge_threshold_unsupported
    self.enable_charge_limits(bat0_up, True)
  File "/build/source/build/../src/linux/integration-test.py", line 297, in enable_charge_limits
    self.dbus.call_sync(UP, device,
gi.repository.GLib.GError: g-io-error-quark: GDBus.Error:org.freedesktop.UPower.GeneralError: Operation is not allowed. (36)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/source/build/../src/linux/integration-test.py", line 1165, in test_battery_charge_threshold_unsupported
    self.assertIn("setting battery charge thresholds", str(err))
AssertionError: 'setting battery charge thresholds' not found in 'g-io-error-quark: GDBus.Error:org.freedesktop.UPower.GeneralError: Operation is not allowed. (36)'

----------------------------------------------------------------------
Ran 1 test in 0.367s

FAILED (failures=1)


[7-17/75] 🌕 Tests.test_battery_zero_power_draw                                                                0/80s
[K[7-18/75] 🌖 Tests.test_battery_zero_power_draw                                                                0/80s
[K 7/75 Tests.test_battery_charge_limit_supported                                                [1;31mFAIL          [0m  1.05s   exit status 1
[1;31m>>> [0mLD_LIBRARY_PATH=/build/source/build/libupower-glib:/build/source/build/libupower-glib MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 GI_TYPELIB_PATH=/build/source/build/libupower-glib:/nix/store/pjxrn2wn0sn533p48jz5qxgjld84hn3i-glib-2.80.4/lib/girepository-1.0:/nix/store/vscjxy29f1w1z36ah69x2v37fvf8p1nm-gobject-introspection-1.80.1/lib/girepository-1.0:/nix/store/m4s69jvb62d9adgj0jxbw5qq3j3b4b9b-libgudev-238/lib/girepository-1.0:/nix/store/5l380ssy468jw4czvjyp25sb739hiczr-polkit-124/lib/girepository-1.0:/nix/store/5ryn154cpg2vw1bn5ybmrcl4mcsjsphq-umockdev-0.18.3/lib/girepository-1.0 MESON_TEST_ITERATION=1 top_builddir=/build/source/build MALLOC_PERTURB_=94 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3 /build/source/build/../src/linux/integration-test.py Tests.test_battery_charge_limit_supported
 ✀  
[1;36mstdout:[0m
Testing binaries from local build tree
1725997137.044 GetAll /org/freedesktop/login1 org.freedesktop.login1.Manager
1725997137.044 Inhibit "sleep" "UPower" "Pause device polling" "delay"
[32mTI:19:38:57	[34mAcquired inhibitor lock (7, delay)
[0m[32mTI:19:38:57	[34mStarting upowerd version 1.90.5
[0m[32mTI:19:38:57	[34mcannot find a keyboard backlight
[0m[32mTI:19:38:57	[34mExported UpDevice with path /org/freedesktop/UPower/devices/DisplayDevice
[0m[32mTI:19:38:57	[34mdaemon now coldplug
[0m[32mTI:19:38:57	[34mregistering subsystem : input
[0m[32mTI:19:38:57	[34mregistering subsystem : power_supply
[0m[32mTI:19:38:57	[34mReceived uevent add on device /sys/devices/BAT0
[0m[32mTI:19:38:57	[34musing present voltage (alternate)
[0m[32mTI:19:38:57	[34mfailed to read battery charge threshold: Failed to open file “/build/upower-history-25z2dlqx/charging-threshold-status”: No such file or directory
[0m[32mTI:19:38:57	[34musing id: test-80
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-25z2dlqx/history-rate-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-25z2dlqx/history-charge-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-25z2dlqx/history-time-full-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34mfailed to get data from /build/upower-history-25z2dlqx/history-time-empty-test-80.dat as file does not exist
[0m[32mTI:19:38:57	[34msaving in 600 seconds
[0m[32mTI:19:38:57	[34mdevice BAT0 has unknown state, not saving history
[0m[32mTI:19:38:57	[34munknown_poll: setting up fast re-poll
[0m[32mTI:19:38:57	[34madded native-path: BAT0
[0m[32mTI:19:38:57	[34mExported UpDevice with path /org/freedesktop/UPower/devices/battery_BAT0
[0m[32mTI:19:38:57	[34mdevice BAT0 has parent id: (null)
[0m[32mTI:19:38:57	[34mGot new device from udev enumerator: 0x45ea50
[0m[32mTI:19:38:57	[34madded BAT0
[0m[32mTI:19:38:57	[34memitting added: /org/freedesktop/UPower/devices/battery_BAT0
[0m[32mTI:19:38:57	[34mregistering subsystem : usb
[0m[32mTI:19:38:57	[34mregistering subsystem : usbmisc
[0m[32mTI:19:38:57	[34mregistering subsystem : sound
[0m[32mTI:19:38:57	[34mon_battery = yes
[0m[32mTI:19:38:57	[34mwarning_level = none
[0m[32mTI:19:38:57	[34mdaemon now not coldplug
[0m[32mTI:19:38:57	[34mBlueZ disappeared
[0m[32mTI:19:38:57	[34mError on Polkit check authority: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.PolicyKit1 was not provided by any .service files
[0m[32mTI:19:38:57	[34mHandling SIGTERM
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-25z2dlqx/history-rate-test-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-25z2dlqx/history-charge-test-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-25z2dlqx/history-time-full-test-80.dat
[0m[32mTI:19:38:57	[34mculled 0 of 1
[0m[32mTI:19:38:57	[34msaved /build/upower-history-25z2dlqx/history-time-empty-test-80.dat
[0m
[1;36mstderr:[0m
E
======================================================================
ERROR: test_battery_charge_limit_supported (__main__.Tests.test_battery_charge_limit_supported)
Battery with charge_control_start/end_threshold supported
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/build/../src/linux/integration-test.py", line 1100, in test_battery_charge_limit_supported
    self.enable_charge_limits(bat0_up, True)
  File "/build/source/build/../src/linux/integration-test.py", line 297, in enable_charge_limits
    self.dbus.call_sync(UP, device,
gi.repository.GLib.GError: g-io-error-quark: GDBus.Error:org.freedesktop.UPower.GeneralError: Operation is not allowed. (36)

----------------------------------------------------------------------
Ran 1 test in 0.388s

FAILED (errors=1)


[8-18/75] 🌗 Tests.test_battery_zero_power_draw                                                                0/80s
[K[8-19/75] 🌘 Tests.test_battery_zero_power_draw                                                                0/80s
[K 8/75 Tests.test_battery_overfull                                                              [1;32mOK            [0m  1.07s
[9-19/75] 🌑 Tests.test_battery_zero_power_draw                                                                0/80s
[K[9-20/75] 🌒 Tests.test_battery_zero_power_draw                                                                0/80s
[K 9/75 Tests.test_battery_energy_charge_mixed                                                   [1;32mOK            [0m  1.08s
[10-20/75] 🌓 Tests.test_battery_zero_power_draw                                                               0/80s
[K[10-21/75] 🌔 Tests.test_battery_zero_power_draw                                                               0/80s
[K10/75 Tests.test_battery_ac                                                                    [1;32mOK            [0m  1.29s
[11-21/75] 🌕 Tests.test_battery_zero_power_draw                                                               0/80s
[K[11-22/75] 🌖 Tests.test_battery_zero_power_draw                                                               0/80s
[K[11-22/75] 🌗 Tests.test_bluetooth_keyboard                                                                    0/80s
[K11/75 Tests.test_battery_temperature                                                           [1;32mOK            [0m  1.03s
[12-22/75] 🌘 Tests.test_bluetooth_keyboard                                                                    0/80s
[K[12-23/75] 🌑 Tests.test_bluetooth_keyboard                                                                    0/80s
[K12/75 Tests.test_bluetooth_hid_mouse_no_legacy_subdevice                                       [1;32mOK            [0m  0.93s
[13-23/75] 🌒 Tests.test_bluetooth_keyboard                                                                    0/80s
[K[13-24/75] 🌓 Tests.test_bluetooth_keyboard                                                                    0/80s
[K13/75 Tests.test_battery_zero_power_draw                                                       [1;32mOK            [0m  1.08s
[14-24/75] 🌔 Tests.test_bluetooth_keyboard                                                                    0/80s
[K[14-25/75] 🌕 Tests.test_bluetooth_keyboard                                                                    0/80s
[K14/75 Tests.test_bluetooth_hid_mouse                                                           [1;32mOK            [0m  1.07s
[15-25/75] 🌖 Tests.test_bluetooth_keyboard                                                                    0/80s
[K[15-26/75] 🌗 Tests.test_bluetooth_keyboard                                                                    0/80s
[K15/75 Tests.test_bluetooth_keyboard                                                            [1;32mOK            [0m  1.00s
[16-26/75] 🌘 Tests.test_bluetooth_mouse_and_keyboard                                                          0/80s
[K[16-27/75] 🌑 Tests.test_bluetooth_mouse_and_keyboard                                                          0/80s
[K16/75 Tests.test_bluetooth_headphones                                                          [1;32mOK            [0m  1.58s
[17-27/75] 🌒 Tests.test_bluetooth_mouse_and_keyboard                                                          0/80s
[K[17-28/75] 🌓 Tests.test_bluetooth_mouse_and_keyboard                                                          0/80s
[K17/75 Tests.test_bluetooth_computer                                                            [1;32mOK            [0m  1.71s
[18-28/75] 🌔 Tests.test_bluetooth_mouse_and_keyboard                                                          0/80s
[K[18-29/75] 🌕 Tests.test_bluetooth_mouse_and_keyboard                                                          0/80s
[K18/75 Tests.test_bluetooth_mouse_and_keyboard                                                  [1;32mOK            [0m  0.63s
[19-29/75] 🌖 Tests.test_bluetooth_phone                                                                       0/80s
[K[19-30/75] 🌗 Tests.test_bluetooth_phone                                                                       0/80s
[K19/75 Tests.test_bluetooth_mouse                                                               [1;32mOK            [0m  0.75s
[20-30/75] 🌘 Tests.test_bluetooth_phone                                                                       0/80s
[K[20-31/75] 🌑 Tests.test_bluetooth_phone                                                                       0/80s
[K20/75 Tests.test_bluetooth_heart_rate_monitor                                                  [1;32mOK            [0m  1.78s
[21-31/75] 🌒 Tests.test_bluetooth_phone                                                                       0/80s
[K[21-32/75] 🌓 Tests.test_bluetooth_phone                                                                       0/80s
[K21/75 Tests.test_battery_id_change                                                             [1;32mOK            [0m  2.97s
[22-32/75] 🌔 Tests.test_bluetooth_phone                                                                       0/80s
[K[22-33/75] 🌕 Tests.test_bluetooth_phone                                                                       0/80s
[K[22-33/75] 🌖 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K22/75 Tests.test_bluetooth_mouse_with_keyboard                                                 [1;32mOK            [0m  0.76s
[23-33/75] 🌗 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K[23-34/75] 🌘 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K23/75 Tests.test_bluetooth_hidpp_mouse                                                         [1;32mOK            [0m  2.29s
[24-34/75] 🌑 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K[24-35/75] 🌒 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K24/75 Tests.test_bluetooth_le_device                                                           [1;32mOK            [0m  1.41s
[25-35/75] 🌓 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K[25-36/75] 🌔 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K25/75 Tests.test_bluetooth_le_mouse                                                            [1;32mOK            [0m  1.41s
[26-36/75] 🌕 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K[26-37/75] 🌖 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K26/75 Tests.test_charge_cycles                                                                 [1;32mOK            [0m  0.74s
[27-37/75] 🌗 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K[27-38/75] 🌘 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                   0/80s
[K[27-38/75] 🌑 Tests.test_daemon_version                                                                        0/80s
[K27/75 Tests.test_bluetooth_wireless_earbuds                                                    [1;32mOK            [0m  1.42s
[28-38/75] 🌒 Tests.test_daemon_version                                                                        0/80s
[K[28-39/75] 🌓 Tests.test_daemon_version                                                                        0/80s
[K28/75 Tests.test_daemon_version                                                                [1;32mOK            [0m  0.63s
[29-39/75] 🌔 Tests.test_display_pending_charge_one_battery                                                    0/80s
[K[29-40/75] 🌕 Tests.test_display_pending_charge_one_battery                                                    0/80s
[K29/75 Tests.test_bluetooth_phone                                                               [1;32mOK            [0m  1.50s
[30-40/75] 🌖 Tests.test_display_pending_charge_one_battery                                                    0/80s
[K[30-41/75] 🌗 Tests.test_display_pending_charge_one_battery                                                    0/80s
[K30/75 Tests.test_display_pending_charge_one_battery                                            [1;32mOK            [0m  0.47s
[31-41/75] 🌘 Tests.test_dup_bluetooth_mouse                                                                   0/80s
[K[31-42/75] 🌑 Tests.test_dup_bluetooth_mouse                                                                   0/80s
[K31/75 Tests.test_display_state_aggregation                                                     [1;32mOK            [0m  0.55s
[32-42/75] 🌒 Tests.test_dup_bluetooth_mouse                                                                   0/80s
[K[32-43/75] 🌓 Tests.test_dup_bluetooth_mouse                                                                   0/80s
[K[32-43/75] 🌔 Tests.test_empty_guessing                                                                        0/80s
[K32/75 self-test                                                                                [1;31mFAIL          [0m  5.15s   killed by signal 5 SIGTRAP
[1;31m>>> [0mLD_LIBRARY_PATH=/build/source/build/libupower-glib MSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 MALLOC_PERTURB_=162 MESON_TEST_ITERATION=1 ASAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1 UBSAN_OPTIONS=halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1 /build/source/build/src/up_self_test
 ✀  
TAP version 13
# random seed: R02Sc762162908ccdc5062d0402f87e2016d
1..6
# Start of power tests
ok 1 /power/backend
ok 2 /power/device
# UPower-DEBUG: added /sys/dummy
# UPower-DEBUG: removed /sys/dummy
ok 3 /power/device_list
# UPower-DEBUG: using id: test
# UPower-DEBUG: failed to get data from /build/upower-test.M5XAEo/history-rate-test.dat as file does not exist
# UPower-DEBUG: failed to get data from /build/upower-test.M5XAEo/history-charge-test.dat as file does not exist
# UPower-DEBUG: failed to get data from /build/upower-test.M5XAEo/history-time-full-test.dat as file does not exist
# UPower-DEBUG: failed to get data from /build/upower-test.M5XAEo/history-time-empty-test.dat as file does not exist
# UPower-DEBUG: saving in 600 seconds
# UPower-DEBUG: limiting data to last 10 seconds
# UPower-DEBUG: length of array (before) 0
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: deferring as earlier timeout is already queued
# UPower-DEBUG: limiting data to last 10 seconds
# UPower-DEBUG: length of array (before) 3
# UPower-DEBUG: limiting data to last 10 seconds
# UPower-DEBUG: length of array (before) 3
# UPower-DEBUG: length of array (after) 2
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-rate-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-charge-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-full-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-empty-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-rate-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-charge-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-full-test.dat
# UPower-DEBUG: culled 0 of 4
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-empty-test.dat
# UPower-DEBUG: using id: test
# UPower-DEBUG: loading 5 items of data from /build/upower-test.M5XAEo/history-rate-test.dat
# UPower-DEBUG: loading 5 items of data from /build/upower-test.M5XAEo/history-charge-test.dat
# UPower-DEBUG: loading 5 items of data from /build/upower-test.M5XAEo/history-time-full-test.dat
# UPower-DEBUG: loading 5 items of data from /build/upower-test.M5XAEo/history-time-empty-test.dat
# UPower-DEBUG: saving in 600 seconds
# UPower-DEBUG: limiting data to last 10 seconds
# UPower-DEBUG: length of array (before) 4
# UPower-DEBUG: culled 3 of 5
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-rate-test.dat
# UPower-DEBUG: culled 3 of 5
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-charge-test.dat
# UPower-DEBUG: culled 3 of 5
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-full-test.dat
# UPower-DEBUG: culled 3 of 5
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-empty-test.dat
# UPower-DEBUG: using id: test
# UPower-DEBUG: loading 3 items of data from /build/upower-test.M5XAEo/history-rate-test.dat
# UPower-DEBUG: loading 3 items of data from /build/upower-test.M5XAEo/history-charge-test.dat
# UPower-DEBUG: loading 3 items of data from /build/upower-test.M5XAEo/history-time-full-test.dat
# UPower-DEBUG: loading 3 items of data from /build/upower-test.M5XAEo/history-time-empty-test.dat
# UPower-DEBUG: saving in 600 seconds
# UPower-DEBUG: limiting data to last 10 seconds
# UPower-DEBUG: length of array (before) 2
# UPower-DEBUG: culled 0 of 3
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-rate-test.dat
# UPower-DEBUG: culled 0 of 3
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-charge-test.dat
# UPower-DEBUG: culled 0 of 3
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-full-test.dat
# UPower-DEBUG: culled 0 of 3
# UPower-DEBUG: saved /build/upower-test.M5XAEo/history-time-empty-test.dat
ok 4 /power/history
# slow test /power/history executed in 5.10 secs
ok 5 /power/native
not ok /power/daemon - UPower-FATAL-ERROR: failed to get polkit authority: Error initializing authority: Could not connect: No such file or directory
Bail out!


[33-43/75] 🌕 Tests.test_empty_guessing                                                                        0/80s
[K[33-44/75] 🌖 Tests.test_empty_guessing                                                                        0/80s
[K33/75 Tests.test_dup_logitech_unifying_usb                                                     [1;32mOK            [0m  0.53s
[34-44/75] 🌗 Tests.test_empty_guessing                                                                        0/80s
[K[34-45/75] 🌘 Tests.test_empty_guessing                                                                        0/80s
[K34/75 Tests.test_daemon_restart                                                                [1;32mOK            [0m  1.78s
[35-45/75] 🌑 Tests.test_empty_guessing                                                                        0/80s
[K[35-46/75] 🌒 Tests.test_empty_guessing                                                                        0/80s
[K35/75 Tests.test_empty_guessing                                                                [1;32mOK            [0m  0.49s
[36-46/75] 🌓 Tests.test_headset_hotplug                                                                       0/80s
[K[36-47/75] 🌔 Tests.test_headset_hotplug                                                                       0/80s
[K36/75 Tests.test_dup_bluetooth_mouse                                                           [1;32mOK            [0m  1.22s
[37-47/75] 🌕 Tests.test_headset_hotplug                                                                       0/80s
[K[37-48/75] 🌖 Tests.test_headset_hotplug                                                                       0/80s
[K37/75 Tests.test_bluetooth_mouse_reconnect                                                     [1;32mOK            [0m  3.19s
[38-48/75] 🌗 Tests.test_headset_hotplug                                                                       0/80s
[K[38-49/75] 🌘 Tests.test_headset_hotplug                                                                       0/80s
[K38/75 Tests.test_full_guessing                                                                 [1;32mOK            [0m  0.65s
[39-49/75] 🌑 Tests.test_headset_hotplug                                                                       0/80s
[K[39-50/75] 🌒 Tests.test_headset_hotplug                                                                       0/80s
[K39/75 Tests.test_headset_detection                                                             [1;32mOK            [0m  0.64s
[40-50/75] 🌓 Tests.test_headset_hotplug                                                                       0/80s
[K[40-51/75] 🌔 Tests.test_headset_hotplug                                                                       0/80s
[K40/75 Tests.test_headset_hotplug                                                               [1;32mOK            [0m  0.75s
[41-51/75] 🌕 Tests.test_ignore_unknown_power_supply_type                                                      0/80s
[K[41-52/75] 🌖 Tests.test_ignore_unknown_power_supply_type                                                      0/80s
[K[41-52/75] 🌗 Tests.test_lib_daemon_properties                                                                 0/80s
[K41/75 Tests.test_headset_wireless_status                                                       [1;32mOK            [0m  0.95s
[42-52/75] 🌘 Tests.test_lib_daemon_properties                                                                 0/80s
[K[42-53/75] 🌑 Tests.test_lib_daemon_properties                                                                 0/80s
[K42/75 Tests.test_hidpp_mouse                                                                   [1;32mOK            [0m  0.75s
[43-53/75] 🌒 Tests.test_lib_daemon_properties                                                                 0/80s
[K[43-54/75] 🌓 Tests.test_lib_daemon_properties                                                                 0/80s
[K43/75 Tests.test_ignore_unknown_power_supply_type                                              [1;32mOK            [0m  0.52s
[44-54/75] 🌔 Tests.test_lib_daemon_properties                                                                 0/80s
[K[44-55/75] 🌕 Tests.test_lib_daemon_properties                                                                 0/80s
[K44/75 Tests.test_lib_daemon_properties                                                         [1;32mOK            [0m  0.53s
[45-55/75] 🌖 Tests.test_macbook_capacity                                                                      0/80s
[K[45-56/75] 🌗 Tests.test_macbook_capacity                                                                      0/80s
[K45/75 Tests.test_lib_up_client_async                                                           [1;32mOK            [0m  0.49s
[46-56/75] 🌘 Tests.test_macbook_capacity                                                                      0/80s
[K[46-57/75] 🌑 Tests.test_macbook_capacity                                                                      0/80s
[K46/75 Tests.test_macbook_capacity                                                              [1;32mOK            [0m  0.54s
[47-57/75] 🌒 Tests.test_map_pending_charge_to_fully_charged                                                   0/80s
[K[47-58/75] 🌓 Tests.test_map_pending_charge_to_fully_charged                                                   0/80s
[K47/75 Tests.test_hidpp_touchpad_race                                                           [1;32mOK            [0m  1.11s
[48-58/75] 🌔 Tests.test_map_pending_charge_to_fully_charged                                                   0/80s
[K[48-59/75] 🌕 Tests.test_map_pending_charge_to_fully_charged                                                   0/80s
[K[48-59/75] 🌖 Tests.test_no_devices                                                                            0/80s
[K48/75 Tests.test_map_pending_charge_to_fully_charged                                           [1;32mOK            [0m  0.59s
[49-59/75] 🌗 Tests.test_no_devices                                                                            0/80s
[K[49-60/75] 🌘 Tests.test_no_devices                                                                            0/80s
[K49/75 Tests.test_no_devices                                                                    [1;32mOK            [0m  0.45s
[50-60/75] 🌑 Tests.test_percentage_low_icon_set                                                               0/80s
[K[50-61/75] 🌒 Tests.test_percentage_low_icon_set                                                               0/80s
[K50/75 Tests.test_multiple_batteries                                                            [1;32mOK            [0m  0.57s
[51-61/75] 🌓 Tests.test_percentage_low_icon_set                                                               0/80s
[K[51-62/75] 🌔 Tests.test_percentage_low_icon_set                                                               0/80s
[K51/75 Tests.test_low_battery_changes_history_save_interval                                     [1;32mOK            [0m  1.52s
[52-62/75] 🌕 Tests.test_percentage_low_icon_set                                                               0/80s
[K[52-63/75] 🌖 Tests.test_percentage_low_icon_set                                                               0/80s
[K52/75 Tests.test_percentage_low_icon_set                                                       [1;32mOK            [0m  0.51s
[53-63/75] 🌗 Tests.test_props_online_ac                                                                       0/80s
[K[53-64/75] 🌘 Tests.test_props_online_ac                                                                       0/80s
[K[53-64/75] 🌑 Tests.test_refresh_after_sleep                                                                   0/80s
[K53/75 Tests.test_props_offline_ac                                                              [1;32mOK            [0m  0.67s
[54-64/75] 🌒 Tests.test_refresh_after_sleep                                                                   0/80s
[K[54-65/75] 🌓 Tests.test_refresh_after_sleep                                                                   0/80s
[K54/75 Tests.test_props_online_ac                                                               [1;32mOK            [0m  1.19s
[55-65/75] 🌔 Tests.test_refresh_after_sleep                                                                   1/80s
[K[55-66/75] 🌕 Tests.test_refresh_after_sleep                                                                   1/80s
[K55/75 Tests.test_refresh_after_sleep                                                           [1;32mOK            [0m  1.16s
[56-66/75] 🌖 Tests.test_sibling_priority_no_overwrite                                                         0/80s
[K[56-67/75] 🌗 Tests.test_sibling_priority_no_overwrite                                                         0/80s
[K[56-67/75] 🌘 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K56/75 Tests.test_hidpp_touchpad                                                                [1;32mOK            [0m  3.71s
[57-67/75] 🌑 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K[57-68/75] 🌒 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K57/75 Tests.test_remove                                                                        [1;32mOK            [0m  1.10s
[58-68/75] 🌓 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K[58-69/75] 🌔 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K58/75 Tests.test_sibling_priority_no_overwrite                                                 [1;32mOK            [0m  0.73s
[59-69/75] 🌕 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K[59-70/75] 🌖 Tests.test_unknown_battery_status_no_ac                                                          0/80s
[K[59-70/75] 🌗 Tests.test_ups_offline_ac                                                                        0/80s
[K59/75 Tests.test_unknown_battery_status_with_ac                                                [1;32mOK            [0m  0.93s
[60-70/75] 🌘 Tests.test_ups_offline_ac                                                                        0/80s
[K[60-71/75] 🌑 Tests.test_ups_offline_ac                                                                        0/80s
[K60/75 Tests.test_ups_no_ac                                                                     [1;32mOK            [0m  0.90s
[61-71/75] 🌒 Tests.test_ups_offline_ac                                                                        0/80s
[K[61-72/75] 🌓 Tests.test_ups_offline_ac                                                                        0/80s
[K61/75 Tests.test_ups_offline_ac                                                                [1;32mOK            [0m  0.61s
[62-72/75] 🌔 Tests.test_vendor_strings                                                                        0/80s
[K[62-73/75] 🌕 Tests.test_vendor_strings                                                                        0/80s
[K62/75 Tests.test_usb_joypad                                                                    [1;32mOK            [0m  0.49s
[63-73/75] 🌖 Tests.test_vendor_strings                                                                        0/80s
[K[63-74/75] 🌗 Tests.test_vendor_strings                                                                        0/80s
[K63/75 Tests.test_vendor_strings                                                                [1;32mOK            [0m  0.52s
[64-74/75] 🌘 Tests.test_wacom_bluetooth                                                                       0/80s
[K[64-75/75] 🌑 Tests.test_wacom_bluetooth                                                                       0/80s
[K64/75 Tests.test_unknown_battery_status_no_ac                                                  [1;32mOK            [0m  1.80s
[65-75/75] 🌒 Tests.test_wacom_bluetooth                                                                       0/80s
[K65/75 Tests.test_virtual_unparented_device                                                     [1;32mOK            [0m  0.49s
[66-75/75] 🌓 Tests.test_wacom_bluetooth                                                                       0/80s
[K[66-75/75] 🌔 Tests.test_wacom_dongle                                                                          0/80s
[K66/75 Tests.test_wacom_bluetooth                                                               [1;32mOK            [0m  1.02s
[67-75/75] 🌕 Tests.test_wacom_dongle                                                                          0/80s
[K[67-75/75] 🌖 Tests.test_prevent_sleep_until_critical_action_is_executed                                       4/80s
[K67/75 Tests.test_wacom_dongle                                                                  [1;32mOK            [0m  1.54s
[68-75/75] 🌗 Tests.test_prevent_sleep_until_critical_action_is_executed                                       4/80s
[K[68-75/75] 🌘 Tests.test_macbook_uevent                                                                        6/80s
[K[68-75/75] 🌑 Tests.test_critical_action_is_taken_repeatedly                                                  10/80s
[K68/75 Tests.test_macbook_uevent                                                                [1;32mOK            [0m  7.51s
[69-75/75] 🌒 Tests.test_critical_action_is_taken_repeatedly                                                  10/80s
[K[69-75/75] 🌓 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_true                 11/80s
[K[69-75/75] 🌔 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_false                12/80s
[K[69-75/75] 🌕 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_false                 14/80s
[K[69-75/75] 🌖 Tests.test_battery_state_guessing                                                               17/80s
[K[69-75/75] 🌗 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true                  16/80s
[K[69-75/75] 🌘 Tests.test_prevent_sleep_until_critical_action_is_executed                                      12/80s
[K[69-75/75] 🌑 Tests.test_critical_action_is_taken_repeatedly                                                  17/80s
[K[69-75/75] 🌒 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_true                 18/80s
[K[69-75/75] 🌓 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_false                19/80s
[K69/75 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_false          [1;32mOK            [0m 21.16s
[70-75/75] 🌔 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_false                20/80s
[K70/75 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_false         [1;32mOK            [0m 20.61s
[71-75/75] 🌕 Tests.test_battery_state_guessing                                                               23/80s
[K71/75 Tests.test_critical_action_is_ignore_and_AllowRiskyCriticalPowerAction_is_true           [1;32mOK            [0m 21.01s
[72-75/75] 🌖 Tests.test_battery_state_guessing                                                               23/80s
[K72/75 Tests.test_critical_action_is_suspend_and_AllowRiskyCriticalPowerAction_is_true          [1;32mOK            [0m 20.60s
[73-75/75] 🌗 Tests.test_battery_state_guessing                                                               23/80s
[K[73-75/75] 🌘 Tests.test_prevent_sleep_until_critical_action_is_executed                                      16/80s
[K[73-75/75] 🌑 Tests.test_critical_action_is_taken_repeatedly                                                  21/80s
[K[73-75/75] 🌒 Tests.test_battery_state_guessing                                                               25/80s
[K[73-75/75] 🌓 Tests.test_prevent_sleep_until_critical_action_is_executed                                      19/80s
[K73/75 Tests.test_prevent_sleep_until_critical_action_is_executed                               [1;32mOK            [0m 20.61s
[74-75/75] 🌔 Tests.test_critical_action_is_taken_repeatedly                                                  24/80s
[K[74-75/75] 🌕 Tests.test_battery_state_guessing                                                               27/80s
[K[74-75/75] 🌖 Tests.test_critical_action_is_taken_repeatedly                                                  25/80s
[K[74-75/75] 🌗 Tests.test_battery_state_guessing                                                               29/80s
[K[74-75/75] 🌘 Tests.test_critical_action_is_taken_repeatedly                                                  27/80s
[K[74-75/75] 🌑 Tests.test_battery_state_guessing                                                               31/80s
[K[74-75/75] 🌒 Tests.test_critical_action_is_taken_repeatedly                                                  29/80s
[K[74-75/75] 🌓 Tests.test_battery_state_guessing                                                               33/80s
[K[74-75/75] 🌔 Tests.test_critical_action_is_taken_repeatedly                                                  31/80s
[K[74-75/75] 🌕 Tests.test_battery_state_guessing                                                               35/80s
[K[74-75/75] 🌖 Tests.test_critical_action_is_taken_repeatedly                                                  33/80s
[K[74-75/75] 🌗 Tests.test_battery_state_guessing                                                               37/80s
[K[74-75/75] 🌘 Tests.test_critical_action_is_taken_repeatedly                                                  35/80s
[K[74-75/75] 🌑 Tests.test_battery_state_guessing                                                               39/80s
[K[74-75/75] 🌒 Tests.test_critical_action_is_taken_repeatedly                                                  37/80s
[K[74-75/75] 🌓 Tests.test_battery_state_guessing                                                               41/80s
[K[74-75/75] 🌔 Tests.test_critical_action_is_taken_repeatedly                                                  39/80s
[K[74-75/75] 🌕 Tests.test_battery_state_guessing                                                               43/80s
[K74/75 Tests.test_critical_action_is_taken_repeatedly                                           [1;32mOK            [0m 41.57s
[75/75] 🌖 Tests.test_battery_state_guessing                                                                  44/80s
[K[75/75] 🌗 Tests.test_battery_state_guessing                                                                  44/80s
[K[75/75] 🌘 Tests.test_battery_state_guessing                                                                  45/80s
[K[75/75] 🌑 Tests.test_battery_state_guessing                                                                  46/80s
[K[75/75] 🌒 Tests.test_battery_state_guessing                                                                  47/80s
[K[75/75] 🌓 Tests.test_battery_state_guessing                                                                  48/80s
[K[75/75] 🌔 Tests.test_battery_state_guessing                                                                  49/80s
[K[75/75] 🌕 Tests.test_battery_state_guessing                                                                  50/80s
[K[75/75] 🌖 Tests.test_battery_state_guessing                                                                  51/80s
[K[75/75] 🌗 Tests.test_battery_state_guessing                                                                  52/80s
[K[75/75] 🌘 Tests.test_battery_state_guessing                                                                  53/80s
[K[75/75] 🌑 Tests.test_battery_state_guessing                                                                  54/80s
[K[75/75] 🌒 Tests.test_battery_state_guessing                                                                  55/80s
[K[75/75] 🌓 Tests.test_battery_state_guessing                                                                  56/80s
[K[75/75] 🌔 Tests.test_battery_state_guessing                                                                  57/80s
[K[75/75] 🌕 Tests.test_battery_state_guessing                                                                  58/80s
[K[75/75] 🌖 Tests.test_battery_state_guessing                                                                  59/80s
[K[75/75] 🌗 Tests.test_battery_state_guessing                                                                  60/80s
[K[75/75] 🌘 Tests.test_battery_state_guessing                                                                  61/80s
[K[75/75] 🌑 Tests.test_battery_state_guessing                                                                  62/80s
[K[75/75] 🌒 Tests.test_battery_state_guessing                                                                  63/80s
[K[75/75] 🌓 Tests.test_battery_state_guessing                                                                  64/80s
[K[75/75] 🌔 Tests.test_battery_state_guessing                                                                  65/80s
[K[75/75] 🌕 Tests.test_battery_state_guessing                                                                  66/80s
[K75/75 Tests.test_battery_state_guessing                                                        [1;32mOK            [0m 67.01s

Summary of Failures:

 1/75 Tests.test_battery_charge_limit_multiple_batteries                               [1;31mFAIL          [0m  0.94s   exit status 1
 6/75 Tests.test_battery_charge_threshold_unsupported                                  [1;31mFAIL          [0m  1.04s   exit status 1
 7/75 Tests.test_battery_charge_limit_supported                                        [1;31mFAIL          [0m  1.05s   exit status 1
32/75 self-test                                                                        [1;31mFAIL          [0m  5.15s   killed by signal 5 SIGTRAP

Ok:                 71  
Expected Fail:      0   
Fail:               4   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   

Full log written to /build/source/build/meson-logs/testlog.txt
```

</details>

Reply:
I got further by enabling polkit in the `testConfig`.

Reply:
I've included the upstream patches that fix the integration tests with polkit.